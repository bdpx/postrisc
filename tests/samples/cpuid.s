.text
    write	"cpuid implemented number"
    alloc	96
    ldi		%r13, 0
    cpuid	%r14, %r13, 0
    write	"cpuid len %x64(r14)"
    write	"cpuid loop"
cpuid_loop:
    cpuid	%r15, %r13, 0
    write	"cpuid[%i64(r13)] = %x64(r15)"
    reps.lt.d	%r13, %r14, cpuid_loop
.end
