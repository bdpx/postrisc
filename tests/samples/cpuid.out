cpuid implemented number
cpuid len 000000000000000b
cpuid loop
cpuid[0] = 000000000000000b
cpuid[1] = 0000000000000001
cpuid[2] = 0000000000000000
cpuid[3] = 0000000080000440
cpuid[4] = 0000000080000440
cpuid[5] = 0000000400000840
cpuid[6] = 0000002000000840
cpuid[7] = 0000000000000040
cpuid[8] = 0000000000000040
cpuid[9] = 0000000000000100
cpuid[10] = 0000000000000014
